tests/test_basic.py

```diff
@@ -64,3 +64,37 @@ def test_convs(devices, extra_options):
         out_ours.norm().backward()
         torch.testing.assert_close(ref_out, out_ours, atol=1e-6, rtol=1e-05)
         torch.testing.assert_close(inputs1.grad, inputs2.grad, atol=1e-5, rtol=1e-05)
+
+
+@pytest.mark.parametrize("devices", [None, ("cpu",), ("cpu", "cpu"), ("cpu", "cpu", "cpu")])
+def test_sharding(devices):
+    for emb_cls in nn.Embedding, nn.EmbeddingBag:
```

Comment:
Why not @pytest.mark.parametrize this one?

Reply:
Done, thanks for the review